### PR TITLE
End of Year Housekeeping

### PR DESCRIPTION
**Pull request by avik-pal**
- [x] Purge Flux from the Tests
- [x] Update Compats
- [x] Static Arrays Type Piracy: https://github.com/JuliaArrays/StaticArrays.jl/pull/1224
  - [ ] Remove SA & SACore from deps
- [x] Fix the Steady State Adjoint Tests
- [x] Remove the Manifest File
- [ ] Purge Flux Remnants in Documentation 
  - [x] Neural ODE Tutorials
  - [x] https://github.com/CarloLucibello/GraphNeuralNetworks.jl/commit/02f9bc5125de1e02c8b288bd766fff41569bb94f

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Handling https://github.com/SciML/SciMLSensitivity.jl/pull/939 and https://github.com/SciML/SciMLSensitivity.jl/pull/940 here too?

**Comment by avik-pal** [comment]
> Handling #939 and #940 here too?

Yes

File: docs/Project.toml [hunk 1]
```diff
@@ -1,23 +1,16 @@
...
 DataInterpolations = "82cc6244-b520-54b8-b5a6-8a565e85f1d0"
-DiffEqBase = "2b5f629d-d688-5b77-993f-72d75c75574e"
+DelayDiffEq = "bcd4f6db-9728-5f36-b5f7-82caef46ccdb"
+DelimitedFiles = "8bb1440f-4735-579b-a4ab-409b98df4dab"
 DiffEqCallbacks = "459566f4-90b8-5000-8ac3-15dfb0a30def"
 DiffEqFlux = "aae7a2af-3d4f-5e19-a356-7da93b79d9d0"
 DiffEqNoiseProcess = "77a26b50-5914-5dd7-bc55-306e6241c503"
-DifferentialEquations = "0c46a032-eb83-5123-abaf-570d42b7fbaa"
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
-Flux = "587475ba-b771-5e3f-ad9e-33799f191a9c"
```
Comment (ChrisRackauckas) [inline docs/Project.toml:12]: We should keep one tutorial with Flux, kind of like we do with SimpleChains. It should be clearly second class citizen though, there mostly to keep alive the example of how to do something with Flux, but just as a "you can do things with Flux too!" but also put a warning on there saying Lux is greatly preferred.

File: Project.toml [hunk 2]
```diff
@@ -61,52 +60,26 @@ FunctionProperties = "0.1"
...
+Random = "<0.0.1, 1"
 RandomNumbers = "1.5.3"
 RecursiveArrayTools = "2.4.2, 3"
 Reexport = "0.2, 1.0"
 ReverseDiff = "1.9"
 SciMLBase = "1.66.0, 2"
 SciMLOperators = "0.1, 0.2, 0.3"
-SimpleNonlinearSolve = "0.1.8"
 SparseDiffTools = "2.5"
+StaticArrays = "1.7.0"
```
Comment (ChrisRackauckas) [inline Project.toml:78]:
```suggestion
StaticArrays = "1.8.0"
```

**Pull request merged**